src/parse/parser.rs

```diff
@@ -78,6 +81,10 @@ impl Input {
         }
     }
 
+    pub(crate) fn previous(&mut self) {
+        self.cursor -= 1;
+    }
+
```

Comment:
I think I would prefer explicit `self.cursor -= 1` instead of `self.previous`. The name just gives me wrong impression.